packages/snap/src/index.ts

```diff
@@ -296,9 +296,14 @@ async function lockedOrInactive() {
     return true;
   }
 
+  const stateManager = await getStateManager();
+
+  // As a side effect, lets sync dev mode
+  // TODO: Consider a better place for this
+  await stateManager.syncDevMode();
```

Comment:
sync dev mode when we check (every minute) if the client is unlocked

we cant access state if the client is locked

we dont want to unlock just to sync dev mode